pkg/kvstore/client.go

```diff
@@ -102,3 +107,40 @@ func Connected() <-chan struct{} {
 	}(c)
 	return c
 }
+
+// GlobalUserMgmtClientPromiseCell provides a promise returning the global kvstore client to perform users
+// management operations, once it has been initialized. Note: client initialization must be handled separately.
+var GlobalUserMgmtClientPromiseCell = cell.Module(
+	"global-kvstore-users-client",
+	"Global KVStore Users Management Client Promise",
+
+	cell.Provide(func(lc hive.Lifecycle) promise.Promise[BackendOperationsUserMgmt] {
+		resolver, promise := promise.New[BackendOperationsUserMgmt]()
+		stop := make(chan struct{})
+		var wg sync.WaitGroup
+
+		lc.Append(hive.Hook{
+			OnStart: func(hive.HookContext) error {
+				wg.Add(1)
+				go func() {
+					select {
+					case <-defaultClientSet:
+						resolver.Resolve(defaultClient)
+					case <-stop:
```

Comment:
You should also handle the case of hive.HookContext being closed, since we can abort startup.

Reply:
I've double checked this, but the `OnStart` context [gets canceled once the Start hooks terminate](https://github.com/cilium/cilium/blob/master/pkg/hive/lifecycle.go#L86-L90), and at that point the etcd client might not yet haven been configured (AFAIU, it [should be used if `OnStart` blocks](https://github.com/cilium/cilium/blob/master/pkg/hive/lifecycle.go#L17-L21), but in this case that is done in a separate goroutine). In case startup is aborted, though, the corresponding `OnStop` method should be called, and that would cause the `stop` channel to be closed and the goroutine to terminate.

Reply:
oops, missed that extra `go`, nevermind!